Feature/service

eintim opened the pull request:
Closes #8 
Closes #7

eintim committed:
Category Service

eintim committed:
Added tag service

eintim committed:
Added category endpoints

eintim committed:
Added tags endpoints

eintim committed:
Merge branch 'dev' into feature/service

eintim committed:
Small fixes

eintim committed:
Added tests

eintim commented:
Two small issues remain...
1. Deleting a non existing resource won't create an error. (tags & category endpoint)
2. It isn't possible to update the name of a tag. To make this possible we need to update both tables (RecipeTag & Tag) at the same time to avoid any issues.

And one db design issue...
Right now every user is able to update/delete any tag. A tag should have a reference to the respective author.

LukasLJL committed:
Important Security Fix

LukasLJL commented:
The first Issue can be fixed like the UserService/delete. 
Maybe we should reconsider how tags should work in our application.

eintim committed:
Fixed Category

eintim committed:
Fixed Tags

eintim committed:
Fixed tests

eintim commented:
Following changes have been made:
- Fixed error messages on deleting non existent tag/category.
- Deleted update Tag Endpoint cause it's we don't really need it.
- Made create/delete Tag Endpoint Admin only. (Users should create new Tags indirectly through Recipe Endpoint and only remove them from their recipes).
- Improved tests.

LukasLJL committed:
Removed Import

LukasLJL committed:
Removed Strange Formatting

LukasLJL reviewed (approved):
lgtm :D

LukasLJL commented:
I fixed some formatting, your IDE has done some strange things, maybe we should fix this for further commits.
